crates/p2p/src/tests/request_response.rs

```diff
@@ -1,50 +1,84 @@
 //! Request-response tests
 
-use std::time::Duration;
-
-use anyhow::bail;
-use tokio::time::sleep;
+use tracing::info;
 
 use super::common::Setup;
-use crate::{commands::Command, events::Event};
+#[cfg(not(feature = "request-response"))]
+use crate::events::GossipEvent;
+use crate::{commands::Command, events::ReqRespEvent};
 
-/// Tests the gossip protocol in an all to all connected network with multiple IDs.
 #[tokio::test(flavor = "multi_thread", worker_threads = 3)]
-async fn request_response_basic() -> anyhow::Result<()> {
-    const USERS_NUM: usize = 2;
-
+async fn test_reqresp_basic() -> anyhow::Result<()> {
     let Setup {
-        mut user_handles,
         cancel,
+        mut user_handles,
         tasks,
-    } = Setup::all_to_all(USERS_NUM).await?;
-
-    let _ = sleep(Duration::from_secs(1)).await;
+    } = Setup::all_to_all(2).await?;
 
+    let req_msg = b"request from node1".to_vec();
+    let resp_msg = b"response from node2".to_vec();
     user_handles[0]
-        .handle
+        .command
         .send_command(Command::RequestMessage {
             peer_id: user_handles[1].peer_id,
-            data: Vec::<u8>::from("Hello, it's a request..."),
+            data: req_msg.clone(),
         })
         .await;
+    info!("Node 1 sent request to Node 2");
+
+    #[cfg(feature = "request-response")]
+    {
+        match user_handles[1].reqresp.next_event().await.unwrap() {
+            ReqRespEvent::ReceivedRequest(data, channel) => {
+                info!("Node 2 received request: {:?}", data);
```

Comment:
Can you please adhere to the following log guidelines: https://www.notion.so/Engineering-Rust-Best-Practices-14c901ba000f8025a974ec8e2a70b4af?source=copy_link#1eb901ba000f802cb2fad1b37ffc82e7

Hence, 

```suggestion
                info!(?data, "Node 2 received request");
```

Please apply this to all the files in `test/`?